provider/cache_test.go

```diff
@@ -77,3 +77,108 @@ func TestCacheReset(t *testing.T) {
 	clock.Add(10*time.Minute + 1)
 	test(3)
 }
+
+// nolint:errcheck
+func TestRetryWithBackoff(t *testing.T) {
+
+	cacheTime := time.Minute * 10
+
+	clock := clock.NewMock()
+	clock.Set(time.Now())
+
+	returnError := true
+
+	functionCalled := false
+
+	g := func() (float64, error) {
+		functionCalled = true
+		if returnError {
+			return float64(1), errors.New("timeout")
+		} else {
+			return float64(1), nil
+		}
+	}
+
+	c := ResettableCached(g, cacheTime)
+	c.clock = clock
+
+	// Put cache in error state
+	c.Get()
+
+	// Half a minute later -> cache time is 10 minutes, so no regular get. No retry because of backoff wait time after first error is 1m.
+	functionCalled = false
+	clock.Add(30 * time.Second)
+	c.Get()
+
+	if functionCalled {
+		t.Errorf("unexpected function call")
+	}
+
+	// Total expired time is now 1m30s. No regular get but retry.
+	clock.Add(60 * time.Second)
```

Comment:
das könntest Du als slice of testcase schreiben, jeder tc hat delta-T und functioncalled